src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageDirectiveFeature.cs

```diff
@@ -11,18 +11,43 @@ public static class PageDirectiveFeature
     {
         public static bool TryGetRouteTemplate(RazorProjectItem projectItem, out string template)
         {
+            if (projectItem == null)
+            {
+                throw new ArgumentNullException(nameof(projectItem));
+            }
+
             const string PageDirective = "@page";
 
+            var stream = projectItem.Read();
```

Comment:
I assume we have a guarantee that the underlying stream for `projectItem.Read` can be read twice?